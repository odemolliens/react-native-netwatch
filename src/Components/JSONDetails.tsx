import * as React from 'react';
import { useContext, useState } from 'react';
import { View, Text, TouchableOpacity, ScrollView, StyleSheet } from 'react-native';
import { Appbar } from 'react-native-paper';
import FeatherIcon from 'react-native-vector-icons/Feather';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

// @ts-ignore
import JSONTree from 'react-native-json-tree';
import { ThemeContext } from '../Theme';

export interface IProps {
  onPressBack: (showJSONDetails: boolean) => void;
  data: any;
  title?: string;
}

export const JSONDetails: React.FC<IProps> = props => {
  const theme = useContext(ThemeContext);
  const [viewJSON, setViewJSON] = useState<boolean>(false);

  const _renderLabel = ([raw]: string) => <Text style={{ fontSize: 14, color: theme.base0D }}>{`${raw} : `}</Text>;

  return (
    <View style={styles.container}>
      <Appbar.Header style={[styles.header, { backgroundColor: theme.secondaryDarkColor }]}>
        <TouchableOpacity
          style={[styles.button, { borderLeftWidth: 0 }]}
          onPress={() => props.onPressBack(false)}
          testID="buttonBackToDetailsScreen"
        >
          <FeatherIcon name="arrow-left" color={theme.textColorOne} size={24} />
        </TouchableOpacity>
        <Appbar.Content color={theme.primaryColor} title={props.title || 'Netwatch'} titleStyle={{ fontSize: 18 }} />
        <TouchableOpacity
          testID="buttonSwitchJSON"
          style={[styles.button, { borderLeftWidth: 0 }]}
          onPress={() => setViewJSON(!viewJSON)}
        >
          <MaterialCommunityIcons
            name="code-json"
            color={viewJSON ? theme.primaryColor : theme.textColorOne}
            size={24}
          />
        </TouchableOpacity>
      </Appbar.Header>

      <ScrollView horizontal style={{ backgroundColor: theme.secondaryColor }}>
        <ScrollView nestedScrollEnabled style={styles.scrollview}>
          {viewJSON ? (
            <JSONTree data={JSON.parse(props.data)} labelRenderer={_renderLabel} theme={theme} invertTheme={false} />
          ) : (
            <Text style={{ color: theme.textColorOne }}>{props.data}</Text>
          )}
        </ScrollView>
      </ScrollView>
    </View>
  );
};

export default JSONDetails;

const styles = StyleSheet.create({
  header: {
    elevation: 0,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0,
  },
  container: {
    position: 'absolute',
    height: '100%',
    width: '100%',
    top: 0,
    left: 0,
  },
  scrollview: {
    paddingHorizontal: 16,
    paddingTop: 20,
    paddingBottom: 20,
  },
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    height: 56,
    width: 56,
    borderLeftWidth: 1,
  },
});
