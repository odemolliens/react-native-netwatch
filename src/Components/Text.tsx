import * as React from 'react';
import { useContext } from 'react';
import { StyleProp, StyleSheet } from 'react-native';
import { Text as RNPText } from 'react-native-paper';
import { ThemeContext } from '../Theme';

export interface IProps {
  children: string;
  style?: StyleProp<any>;
  numberOfLines?: number;
}

export const Text = (props: IProps) => {
  const theme = useContext(ThemeContext);

  return (
    <RNPText style={[styles.text, { color: theme.textColorOne }, props.style]} {...props}>
      {props.children}
    </RNPText>
  );
};

export const TextSecondaryColor = (props: IProps) => {
  const theme = useContext(ThemeContext);

  return (
    <RNPText style={[styles.text, { color: theme.textColorTwo }, props.style]} {...props}>
      {props.children}
    </RNPText>
  );
};

export const Title = (props: IProps) => {
  const theme = useContext(ThemeContext);

  return (
    <RNPText style={[styles.text, styles.text, { color: theme.textColorOne }, props.style]} {...props}>
      {props.children}
    </RNPText>
  );
};

const styles = StyleSheet.create({
  text: {
    fontSize: 12,
    lineHeight: 20,
  },
  textBold: {
    fontWeight: 'bold',
  },
  subtitle: {
    fontSize: 14,
    lineHeight: 22,
  },
  title: {
    fontSize: 16,
    lineHeight: 24,
  },
});
