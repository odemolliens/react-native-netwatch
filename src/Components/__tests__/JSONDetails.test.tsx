import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { JSONDetails, IProps } from '../JSONDetails';
// Need to use this lib to render child correctly
import { render } from '@testing-library/react-native';
import JSONTree from 'react-native-json-tree';
import { Text } from '../Text';

describe('Main test suite', () => {
  let component: ShallowWrapper;
  let props: IProps;
  const onPressBack = jest.fn();
  const setViewJSON = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render properly without title', () => {
    givenProps(stringData);
    const { toJSON } = render(<JSONDetails {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render properly in raw', () => {
    givenProps(stringData, 'JSONDetails');
    givenProps(stringData);
    const { toJSON } = render(<JSONDetails {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render properly in JSON', () => {
    const useStateMock: any = (viewJSON: any) => [true, setViewJSON];
    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
    givenProps(stringData, 'JSONDetails');
    givenProps(stringData);
    const { toJSON } = render(<JSONDetails {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should called setViewJSON', () => {
    const useStateMock: any = (viewJSON: any) => [viewJSON, setViewJSON];
    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
    givenProps(stringData);
    givenComponent();
    whenPressingButton('buttonSwitchJSON');
    expect(setViewJSON).toHaveBeenCalledTimes(1);
    component.update();
    expect(component.find(Text)).toHaveLength(1);
    expect(component.find(JSONTree)).toHaveLength(0);
  });

  it('should display raw Text if viewJSON is false', () => {
    const useStateMock: any = (viewJSON: any) => [false, setViewJSON];
    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
    givenProps(stringData);
    givenComponent();
    expect(component.find(Text)).toHaveLength(1);
    expect(component.find(JSONTree)).toHaveLength(0);
  });

  it('should display raw JSONTree if viewJSON is true', () => {
    const useStateMock: any = (viewJSON: any) => [true, setViewJSON];
    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
    givenProps(stringData);
    givenComponent();
    expect(component.find(Text)).toHaveLength(0);
    expect(component.find(JSONTree)).toHaveLength(1);
  });

  it('should called setViewJSON', () => {
    const useStateMock: any = (viewJSON: any) => [viewJSON, setViewJSON];
    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
    givenProps(stringData);
    givenComponent();
    whenPressingButton('buttonSwitchJSON');
    expect(setViewJSON).toHaveBeenCalledTimes(1);
    component.update();
    expect(component.find(Text)).toHaveLength(1);
    expect(component.find(JSONTree)).toHaveLength(0);
  });

  it('should render properly and go back', () => {
    givenProps(stringData, 'JSONDetails');
    givenComponent();
    whenPressingButton('buttonBackToDetailsScreen');
    expect(props.onPressBack).toHaveBeenCalledTimes(1);
    expect(props.onPressBack).toHaveBeenCalledWith(false);
  });

  // UTILITIES
  function whenPressingButton(testId: string) {
    component.find(`[testID="${testId}"]`).simulate('press');
  }

  // GIVEN
  function givenComponent() {
    component = shallow(<JSONDetails {...props} />);
  }

  function givenProps(data: string = '{}', title?: string) {
    props = {
      onPressBack,
      data,
      title,
    };
  }
});

const stringData = JSON.stringify({
  args: {
    query: 'some really long query that goes onto multiple lines so we can test what happens',
  },
  data: {
    test: 'hello',
  },
  files: {},
  form: {},
  headers: {
    'x-forwarded-proto': 'https',
    'x-forwarded-port': '443',
    host: 'postman-echo.com',
    'x-amzn-trace-id': 'Root=1-605871d5-7bf937ce7fd0d4c0392726bc',
    'content-length': '16',
    'content-type': 'application/json',
    accept: '*/*',
    'x-special': 'dev-test',
    'if-none-match': 'W/"32e-r/l1NhNWVizxAiJazd2G/2G3rHs"',
    'accept-language': 'en-us',
    'accept-encoding': 'gzip, deflate, br',
    'user-agent': 'example/1 CFNetwork/1220.1 Darwin/20.3.0',
    'x-others':
      'Another test with a very very long string just to see how it is render in the application and if it is all done',
    cookie: 'sails.sid=s%3A9fHegW1j43POAQ8xKlXTuE7wfhWp9guw.ZM4PklZGQHdDSAeo7%2FqvaE3LjGdLyf9KcAuv6cK0hpY',
  },
  json: {
    test: 'hello',
  },
  url:
    'https://postman-echo.com/post?query=some%20really%20long%20query%20that%20goes%20onto%20multiple%20lines%20so%20we%20can%20test%20what%20happens',
});
