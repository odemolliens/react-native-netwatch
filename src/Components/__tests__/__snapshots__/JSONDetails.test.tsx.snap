// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main test suite should render properly in JSON 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    horizontal={true}
    style={
      Object {
        "backgroundColor": "#1F2937",
      }
    }
  >
    <View>
      <RCTScrollView
        nestedScrollEnabled={true}
        style={
          Object {
            "paddingBottom": 20,
            "paddingHorizontal": 16,
            "paddingTop": 20,
          }
        }
      >
        <View>
          <View
            className=""
            style={
              Object {
                "backgroundColor": "#1F2937",
                "padding": 5,
              }
            }
          >
            <View
              className=""
              style={
                Object {
                  "marginLeft": 0,
                  "paddingBottom": 5,
                  "paddingLeft": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    className=""
                    style={
                      Object {
                        "flexDirection": "row",
                        "paddingLeft": 0,
                        "paddingRight": 3,
                      }
                    }
                  >
                    <Text
                      className=""
                      style={
                        Object {
                          "color": "#F9FAFB",
                          "fontSize": 15,
                          "marginLeft": 0,
                          "transform": Array [
                            Object {
                              "rotate": "90deg",
                            },
                          ],
                        }
                      }
                    >
                      ▶
                    </Text>
                  </View>
                </View>
                <Text
                  className=""
                  onPress={[Function]}
                  style={
                    Object {
                      "color": "#F9FAFB",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#F9FAFB",
                        "fontSize": 14,
                      }
                    }
                  >
                    root : 
                  </Text>
                </Text>
                <Text
                  className=""
                  onPress={[Function]}
                  style={
                    Object {
                      "color": "#06B6D4",
                      "marginLeft": 5,
                    }
                  }
                >
                  <Text>
                    <Text
                      className=""
                      style={Object {}}
                    >
                      {}
                    </Text>
                     
                    7 keys
                  </Text>
                </Text>
              </View>
              <View
                className=""
                style={
                  Object {
                    "marginLeft": 10,
                  }
                }
              >
                <View
                  className=""
                  style={
                    Object {
                      "marginLeft": 7,
                      "paddingBottom": 5,
                      "paddingLeft": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        className=""
                        style={
                          Object {
                            "flexDirection": "row",
                            "paddingLeft": 0,
                            "paddingRight": 3,
                          }
                        }
                      >
                        <Text
                          className=""
                          style={
                            Object {
                              "color": "#F9FAFB",
                              "fontSize": 15,
                              "marginLeft": 0,
                              "transform": Array [
                                Object {
                                  "rotate": "0deg",
                                },
                              ],
                            }
                          }
                        >
                          ▶
                        </Text>
                      </View>
                    </View>
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#F9FAFB",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#F9FAFB",
                            "fontSize": 14,
                          }
                        }
                      >
                        args : 
                      </Text>
                    </Text>
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#4D7C0F",
                          "marginLeft": 5,
                        }
                      }
                    >
                      <Text>
                        <Text
                          className=""
                          style={Object {}}
                        >
                          {}
                        </Text>
                         
                        1 key
                      </Text>
                    </Text>
                  </View>
                </View>
                <View
                  className=""
                  style={
                    Object {
                      "marginLeft": 7,
                      "paddingBottom": 5,
                      "paddingLeft": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        className=""
                        style={
                          Object {
                            "flexDirection": "row",
                            "paddingLeft": 0,
                            "paddingRight": 3,
                          }
                        }
                      >
                        <Text
                          className=""
                          style={
                            Object {
                              "color": "#F9FAFB",
                              "fontSize": 15,
                              "marginLeft": 0,
                              "transform": Array [
                                Object {
                                  "rotate": "0deg",
                                },
                              ],
                            }
                          }
                        >
                          ▶
                        </Text>
                      </View>
                    </View>
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#F9FAFB",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#F9FAFB",
                            "fontSize": 14,
                          }
                        }
                      >
                        data : 
                      </Text>
                    </Text>
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#4D7C0F",
                          "marginLeft": 5,
                        }
                      }
                    >
                      <Text>
                        <Text
                          className=""
                          style={Object {}}
                        >
                          {}
                        </Text>
                         
                        1 key
                      </Text>
                    </Text>
                  </View>
                </View>
                <View
                  className=""
                  style={
                    Object {
                      "marginLeft": 7,
                      "paddingBottom": 5,
                      "paddingLeft": 3,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#F9FAFB",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#F9FAFB",
                            "fontSize": 14,
                          }
                        }
                      >
                        files : 
                      </Text>
                    </Text>
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#4D7C0F",
                          "marginLeft": 5,
                        }
                      }
                    >
                      <Text>
                        <Text
                          className=""
                          style={Object {}}
                        >
                          {}
                        </Text>
                         
                        0 keys
                      </Text>
                    </Text>
                  </View>
                </View>
                <View
                  className=""
                  style={
                    Object {
                      "marginLeft": 7,
                      "paddingBottom": 5,
                      "paddingLeft": 3,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#F9FAFB",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#F9FAFB",
                            "fontSize": 14,
                          }
                        }
                      >
                        form : 
                      </Text>
                    </Text>
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#4D7C0F",
                          "marginLeft": 5,
                        }
                      }
                    >
                      <Text>
                        <Text
                          className=""
                          style={Object {}}
                        >
                          {}
                        </Text>
                         
                        0 keys
                      </Text>
                    </Text>
                  </View>
                </View>
                <View
                  className=""
                  style={
                    Object {
                      "marginLeft": 7,
                      "paddingBottom": 5,
                      "paddingLeft": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        className=""
                        style={
                          Object {
                            "flexDirection": "row",
                            "paddingLeft": 0,
                            "paddingRight": 3,
                          }
                        }
                      >
                        <Text
                          className=""
                          style={
                            Object {
                              "color": "#F9FAFB",
                              "fontSize": 15,
                              "marginLeft": 0,
                              "transform": Array [
                                Object {
                                  "rotate": "0deg",
                                },
                              ],
                            }
                          }
                        >
                          ▶
                        </Text>
                      </View>
                    </View>
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#F9FAFB",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#F9FAFB",
                            "fontSize": 14,
                          }
                        }
                      >
                        headers : 
                      </Text>
                    </Text>
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#4D7C0F",
                          "marginLeft": 5,
                        }
                      }
                    >
                      <Text>
                        <Text
                          className=""
                          style={Object {}}
                        >
                          {}
                        </Text>
                         
                        14 keys
                      </Text>
                    </Text>
                  </View>
                </View>
                <View
                  className=""
                  style={
                    Object {
                      "marginLeft": 7,
                      "paddingBottom": 5,
                      "paddingLeft": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        className=""
                        style={
                          Object {
                            "flexDirection": "row",
                            "paddingLeft": 0,
                            "paddingRight": 3,
                          }
                        }
                      >
                        <Text
                          className=""
                          style={
                            Object {
                              "color": "#F9FAFB",
                              "fontSize": 15,
                              "marginLeft": 0,
                              "transform": Array [
                                Object {
                                  "rotate": "0deg",
                                },
                              ],
                            }
                          }
                        >
                          ▶
                        </Text>
                      </View>
                    </View>
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#F9FAFB",
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#F9FAFB",
                            "fontSize": 14,
                          }
                        }
                      >
                        json : 
                      </Text>
                    </Text>
                    <Text
                      className=""
                      onPress={[Function]}
                      style={
                        Object {
                          "color": "#4D7C0F",
                          "marginLeft": 5,
                        }
                      }
                    >
                      <Text>
                        <Text
                          className=""
                          style={Object {}}
                        >
                          {}
                        </Text>
                         
                        1 key
                      </Text>
                    </Text>
                  </View>
                </View>
                <View
                  className=""
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginLeft": 10,
                    }
                  }
                >
                  <Text
                    className=""
                    style={
                      Object {
                        "color": "#F9FAFB",
                        "flexDirection": "row",
                        "marginBottom": 8,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#F9FAFB",
                          "fontSize": 14,
                        }
                      }
                    >
                      url : 
                    </Text>
                  </Text>
                  <Text
                    className=""
                    style={
                      Object {
                        "color": "#4D7C0F",
                        "marginLeft": 5,
                      }
                    }
                  >
                    "https://postman-echo.com/post?query=some%20really%20long%20query%20that%20goes%20onto%20multiple%20lines%20so%20we%20can%20test%20what%20happens"
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Main test suite should render properly in raw 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    horizontal={true}
    style={
      Object {
        "backgroundColor": "#1F2937",
      }
    }
  >
    <View>
      <RCTScrollView
        nestedScrollEnabled={true}
        style={
          Object {
            "paddingBottom": 20,
            "paddingHorizontal": 16,
            "paddingTop": 20,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#F9FAFB",
              }
            }
          >
            {"args":{"query":"some really long query that goes onto multiple lines so we can test what happens"},"data":{"test":"hello"},"files":{},"form":{},"headers":{"x-forwarded-proto":"https","x-forwarded-port":"443","host":"postman-echo.com","x-amzn-trace-id":"Root=1-605871d5-7bf937ce7fd0d4c0392726bc","content-length":"16","content-type":"application/json","accept":"*/*","x-special":"dev-test","if-none-match":"W/\\"32e-r/l1NhNWVizxAiJazd2G/2G3rHs\\"","accept-language":"en-us","accept-encoding":"gzip, deflate, br","user-agent":"example/1 CFNetwork/1220.1 Darwin/20.3.0","x-others":"Another test with a very very long string just to see how it is render in the application and if it is all done","cookie":"sails.sid=s%3A9fHegW1j43POAQ8xKlXTuE7wfhWp9guw.ZM4PklZGQHdDSAeo7%2FqvaE3LjGdLyf9KcAuv6cK0hpY"},"json":{"test":"hello"},"url":"https://postman-echo.com/post?query=some%20really%20long%20query%20that%20goes%20onto%20multiple%20lines%20so%20we%20can%20test%20what%20happens"}
          </Text>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Main test suite should render properly without title 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    horizontal={true}
    style={
      Object {
        "backgroundColor": "#1F2937",
      }
    }
  >
    <View>
      <RCTScrollView
        nestedScrollEnabled={true}
        style={
          Object {
            "paddingBottom": 20,
            "paddingHorizontal": 16,
            "paddingTop": 20,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#F9FAFB",
              }
            }
          >
            {"args":{"query":"some really long query that goes onto multiple lines so we can test what happens"},"data":{"test":"hello"},"files":{},"form":{},"headers":{"x-forwarded-proto":"https","x-forwarded-port":"443","host":"postman-echo.com","x-amzn-trace-id":"Root=1-605871d5-7bf937ce7fd0d4c0392726bc","content-length":"16","content-type":"application/json","accept":"*/*","x-special":"dev-test","if-none-match":"W/\\"32e-r/l1NhNWVizxAiJazd2G/2G3rHs\\"","accept-language":"en-us","accept-encoding":"gzip, deflate, br","user-agent":"example/1 CFNetwork/1220.1 Darwin/20.3.0","x-others":"Another test with a very very long string just to see how it is render in the application and if it is all done","cookie":"sails.sid=s%3A9fHegW1j43POAQ8xKlXTuE7wfhWp9guw.ZM4PklZGQHdDSAeo7%2FqvaE3LjGdLyf9KcAuv6cK0hpY"},"json":{"test":"hello"},"url":"https://postman-echo.com/post?query=some%20really%20long%20query%20that%20goes%20onto%20multiple%20lines%20so%20we%20can%20test%20what%20happens"}
          </Text>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
